helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -279,6 +409,16 @@ public Object removeRoutingTableChangeListener(
     return getInstancesForResource(resourceName, partitionName, state);
   }
 
+  public List<InstanceConfig> getInstances(String resourceName, String partitionName, String state,
```

Comment:
Are we support customize state reverse search? If not in the design, let's hold this for now.

Reply:
Removed.

Reply:
Now removed.